src/modules/sndio.cpp

```diff
@@ -140,27 +135,27 @@ auto Sndio::put_val(unsigned int addr, unsigned int val) -> void {
   }
 }
 
-bool Sndio::handleScroll(GdkEventScroll *e) {
+bool Sndio::handleScroll(double dx, double dy) {
   // change the volume only when no user provided
   // events are configured
   if (config_["on-scroll-up"].isString() || config_["on-scroll-down"].isString()) {
-    return AModule::handleScroll(e);
+    return AModule::handleScroll(dx, dy);
   }
 
   // only try to talk to sndio if connected
   if (hdl_ == nullptr) return true;
 
-  auto dir = AModule::getScrollDir(e);
+  auto dir{AModule::getScrollDir(controllScroll_->get_current_event())};
   if (dir == SCROLL_DIR::NONE) {
     return true;
   }
 
-  int step = 5;
+  int step{5};
   if (config_["scroll-step"].isInt()) {
```

Comment:
Does it ever make sense for this to be negative? If not, we could read this as a UInt, and avoid some casting